### ISSUE TITLE
Maximum call stack size exceeded with update from 2.2.3 to 2.3.1

### ISSUE THREAD

**Issue by cpauwels** [unjs/nitro#1063]
### Environment

* Nitro 2.3.1
* Node 18.14.0
* Cloudflare Pages preset

### Reproduction

All my projects are failing with the same error, but when I scaffold a brand new project the issue is not present. I haven't been able to isolate the issue.

### Describe the bug

When building the project with Nuxt 3.3.1 and Nitro 2.3.1 I get a `ERROR  Maximum call stack size exceeded`.

Building and deploying the project with previous Nuxt and Nitro versions worked flawlessly. I'm guessing it's a regression due to the new Cloudflare Pages improvements from #1004 ?

### Logs

```sh
Nuxi 3.3.1                                                                                                                                                                                               17:09:51
Nuxt 3.3.1 with Nitro 2.3.1                                                                                                                                                                              17:09:51
ℹ Using Tailwind CSS from ~/assets/css/tailwind.css                                                                                                                                     nuxt:tailwindcss 17:09:51
ℹ Building client...                                                                                                                                                                                     17:09:52
ℹ vite v4.1.4 building for production...                                                                                                                                                                 17:09:52
ℹ ✓ 1696 modules transformed.                                                                                                                                                                            17:09:59

 ERROR  Maximum call stack size exceeded                                                                                                                                                                 17:09:59


 ERROR  Maximum call stack size exceeded                                                                                                                                                                 17:09:59

  at CallExpression.deoptimizePath (node_modules/rollup/dist/es/shared/node-entry.js:9575:19)
  at LogicalExpression.deoptimizePath (node_modules/rollup/dist/es/shared/node-entry.js:11246:24)
  at ArrowFunctionExpression.deoptimizePath (node_modules/rollup/dist/es/shared/node-entry.js:8605:46)
  at ObjectEntity.deoptimizeArgumentsOnInteractionAtPath (node_modules/rollup/dist/es/shared/node-entry.js:6054:30)
  at ObjectExpression.deoptimizeArgumentsOnInteractionAtPath (node_modules/rollup/dist/es/shared/node-entry.js:11529:32)
  at node_modules/rollup/dist/es/shared/node-entry.js:6638:83
  at PathTracker.withTrackedEntityAtPath (node_modules/rollup/dist/es/shared/node-entry.js:1784:24)
  at LocalVariable.deoptimizeArgumentsOnInteractionAtPath (node_modules/rollup/dist/es/shared/node-entry.js:6638:26)
  at Identifier.deoptimizeArgumentsOnInteractionAtPath (node_modules/rollup/dist/es/shared/node-entry.js:8141:23)
  at ParameterVariable.addEntityToBeDeoptimized (node_modules/rollup/dist/es/shared/node-entry.js:6779:20)
```

**Comment by cpauwels** [comment]
After some more digging, the issue seems to come from the latest (3.19) release of https://github.com/rollup/rollup/issues/4904

**Comment by danielroe** [comment]
This issue has now been resolved in rollup@3.20.3 - you can refresh your lockfile to pull it in.
